src/custom/hooks/usePriceImpact/index.ts

```diff
@@ -123,24 +132,41 @@ interface AbaPriceImpactParams {
   fiatPriceImpact?: Percent
 }
 
-function _calculateAbaPriceImpact(abTrade: TradeGp, baTrade: TradeGp, isExactIn: boolean) {
-  console.debug('abTrade', abTrade, 'baTrade', baTrade, isExactIn)
+function _calculateAbaPriceImpact(initialValue: string, finalValue: string) {
+  const initialValueBn = new BigNumber(initialValue)
+  const finalValueBn = new BigNumber(finalValue)
+  // TODO: use correct formula
+  // ((IV - FV) / IV / 2) * 100
+  const [numerator, denominator] = initialValueBn.minus(finalValueBn).div(initialValueBn).div('2').toFraction()
+
+  // we don't want zeroes
+  if (numerator.isZero() || denominator.isZero()) return undefined
+
+  const priceImpactPercentage =
+    // Uni sdk hates negative numbers so we need to do this
+    numerator.isNegative() || denominator.isNegative()
+      ? new Percent(numerator.absoluteValue().toString(10), denominator.absoluteValue().toString(10)).multiply('-1')
+      : new Percent(numerator.toString(10), denominator.toString(10))
 
-  return HARD_CODED_PRICE_IMPACT
+  console.debug(`[useAbaPriceImpact]::${priceImpactPercentage.toSignificant(2)}%`)
+
+  return priceImpactPercentage
 }
 
 export function useAbaPriceImpact({ abTrade, fiatPriceImpact }: AbaPriceImpactParams) {
```

Comment:
Should we have each hook in a file?  Now that is a folder maybe this would reduce the index.ts a lot

Also, I like you defined types for the params. Should we do the same for `useQuoteAndSwap`